src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/GroupCollection.cs

```diff
@@ -250,5 +250,69 @@ void IEnumerator.Reset()
 
             void IDisposable.Dispose() { }
         }
+
+        public bool TryGetValue(string key, out Group value)
+        {
+            if (ContainsKey(key))
+            {
+                value = this[key];
```

Comment:
Is this going to incur two lookups, one for ContainsKey and one for the indexer?  I'm not familiar with the costs of those operations (@ViktorHofer?), but I'm wondering if the implementation would instead be better something like:
```C#
public bool TryGetValue(string key, out Group value)
{
    Group g = this[key];
    if (g != Group.s_emptyGroup)
    {
        value = g;
        return true;
    }

    value = null;
    return false;
}
```

Reply:
It's better to change the existing data structure from a Hashtable to a Dictionary<,>. As this type is not serializable we are not breaking anything. http://source.dot.net/#System.Text.RegularExpressions/System/Text/RegularExpressions/GroupCollection.cs,23

Reply:
They way i implemented ContainsKey is that it doesn't do a full lookup of the value, it really only checks if the name exists (if it can resolve an index from GroupNumberFromName)

I'll see if i can make it a bit more efficent though